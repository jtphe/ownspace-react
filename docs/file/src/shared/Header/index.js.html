<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <base data-ice="baseUrl" href="../../../../">
  <title data-ice="title">src/shared/Header/index.js</title>
  <link type="text/css" rel="stylesheet" href="css/style.css">
  <link type="text/css" rel="stylesheet" href="css/prettify-tomorrow.css">
  <script src="script/prettify/prettify.js"></script>
  <script src="script/manual.js"></script>
</head>
<body class="layout-container" data-ice="rootContainer">

<header>
  <a href="./">Home</a>
  
  <a href="identifiers.html">Reference</a>
  <a href="source.html">Source</a>
  
  <div class="search-box">
  <span>
    <img src="./image/search.png">
    <span class="search-input-edge"></span><input class="search-input"><span class="search-input-edge"></span>
  </span>
    <ul class="search-result"></ul>
  </div>
</header>

<nav class="navigation" data-ice="nav"><div>
  <ul>
    
  <li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#api">api</a><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-api">api</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-home">components/Home</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Home">Home</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-MenuPlus">MenuPlus</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-login">components/Login</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-GenerateTotp">GenerateTotp</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Login">Login</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-SecondAuthScreen">SecondAuthScreen</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-TotpAuthScreen">TotpAuthScreen</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-menu">components/Menu</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-IconMenu">IconMenu</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-NavMenu">NavMenu</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#constants">constants</a><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-RECURRING_ROLE">RECURRING_ROLE</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-ROLE_ONE">ROLE_ONE</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-ROLE_THREE">ROLE_THREE</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-ROLE_TWO">ROLE_TWO</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#router">router</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-AppRouter">AppRouter</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-avatar">shared/Avatar</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Avatar">Avatar</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-clienttext">shared/ClientText</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-TextClient">TextClient</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-custombutton">shared/CustomButton</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-CustomButton">CustomButton</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-header">shared/Header</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Header">Header</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-WithBack">WithBack</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-logo">shared/Logo</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Logo">Logo</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-progressbar">shared/ProgressBar</a><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/shared/ProgressBar/index.js~Animations.html">Animations</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-text">shared/Text</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-TextHelvetica">TextHelvetica</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#utils">utils</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-showToast">showToast</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-mimeType">mimeType</a></span></span></li>
</ul>
</div>
</nav>

<div class="content" data-ice="content"><h1 data-ice="title">src/shared/Header/index.js</h1>
<pre class="source-code line-number raw-source-code"><code class="prettyprint linenums" data-ice="content">/* eslint-disable global-require */
import React from &apos;react&apos;;
import { View, Image, StyleSheet } from &apos;react-native&apos;;
import { ifIphoneX } from &apos;react-native-iphone-x-helper&apos;;

/**
 * The Header component
 */
const Header = () =&gt; {
  /**
   * Render the Header component
   * @returns {React.Component} - Header component
   */
  return (
    &lt;View style={styles.container}&gt;
      &lt;Image
        source={require(&apos;./cci_logo.png&apos;)}
        style={styles.logo}
        resizeMode=&quot;contain&quot;
      /&gt;
    &lt;/View&gt;
  );
};

/**
 * Styles of Header component
 */
const styles = StyleSheet.create({
  container: {
    backgroundColor: &apos;#003466&apos;,
    paddingBottom: 30,
    ...ifIphoneX(
      {
        paddingTop: 60
      },
      {
        paddingTop: 40
      }
    )
  },
  logo: {
    alignSelf: &apos;center&apos;,
    marginLeft: -10,
    resizeMode: &apos;contain&apos;,
    width: 50,
    height: 50
  }
});

export default Header;
</code></pre>

</div>

<footer class="footer">
  Generated by <a href="https://esdoc.org">ESDoc<span data-ice="esdocVersion">(1.1.0)</span><img src="./image/esdoc-logo-mini-black.png"></a>
</footer>

<script src="script/search_index.js"></script>
<script src="script/search.js"></script>
<script src="script/pretty-print.js"></script>
<script src="script/inherited-summary.js"></script>
<script src="script/test-summary.js"></script>
<script src="script/inner-link.js"></script>
<script src="script/patch-for-local.js"></script>
</body>
</html>
