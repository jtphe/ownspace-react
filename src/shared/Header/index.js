/* eslint-disable global-require */
import React from 'react';
import { View, Image, StyleSheet } from 'react-native';
import { ifIphoneX } from 'react-native-iphone-x-helper';

const Header = () => {
  return (
    <View style={styles.container}>
      <Image
        source={require('./cci_logo.png')}
        style={styles.logo}
        resizeMode="contain"
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#003466',
    paddingBottom: 30,
    ...ifIphoneX(
      {
        paddingTop: 60
      },
      {
        paddingTop: 40
      }
    )
  },
  logo: {
    alignSelf: 'center',
    marginLeft: -10,
    resizeMode: 'contain',
    width: 50,
    height: 50
  }
});

export default Header;
