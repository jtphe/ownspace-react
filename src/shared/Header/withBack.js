/* eslint-disable global-require */
import React from 'react';
import { View, Image, StyleSheet } from 'react-native';
import { ifIphoneX } from 'react-native-iphone-x-helper';
import Icon from 'react-native-vector-icons/Ionicons';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { Actions } from 'react-native-router-flux';

const WithBack = ({ goTo }) => {
  return (
    <View style={styles.container}>
      <TouchableOpacity style={styles.backContainer} onPress={goTo}>
        <Icon name="ios-arrow-back" size={38} color="white" />
      </TouchableOpacity>
      <View style={styles.logoContainer}>
        <Image
          source={require('./cci_logo.png')}
          style={styles.logo}
          resizeMode="contain"
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#003466',
    paddingBottom: 30,
    ...ifIphoneX(
      {
        paddingTop: 60
      },
      {
        paddingTop: 40
      }
    )
  },
  logo: {
    width: 50,
    height: 50
  },
  logoContainer: {
    alignItems: 'center',
    marginLeft: 136
  },
  backContainer: { marginLeft: 30, marginTop: 8 }
});

export default WithBack;
