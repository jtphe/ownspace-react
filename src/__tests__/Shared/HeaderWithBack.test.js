/* eslint-disable global-require */
import { Image, View } from 'react-native';
import React from 'react';
import Header from '@shared/Header/withBack';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ifIphoneX } from 'react-native-iphone-x-helper';
import { CLIENT_COLOR_PRIMARY } from '@constants';
import { TouchableOpacity } from 'react-native-gesture-handler';

describe('Header tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  it('Header snapshot', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Logo header', () => {
    const logo = wrapper.find(Image).get(0);
    expect(logo.props.source).toEqual(
      require('../../shared/Header/cci_logo.png')
    );
    expect(logo.props.style).toEqual({
      width: 50,
      height: 50
    });
    expect(logo.props.resizeMode).toEqual('contain');
  });

  it('Icon back', () => {
    const iconBack = wrapper.find(TouchableOpacity).get(0).props.children;
    expect(iconBack.props.name).toEqual('ios-arrow-back');
    expect(iconBack.props.size).toEqual(38);
    expect(iconBack.props.color).toEqual('white');
    expect(iconBack.props.allowFontScaling).toEqual(false);
  });

  it('Header style', () => {
    const header = wrapper.find(View).get(0);
    expect(header.props.style).toEqual({
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: CLIENT_COLOR_PRIMARY,
      paddingBottom: 30,
      ...ifIphoneX(
        {
          paddingTop: 60
        },
        {
          paddingTop: 40
        }
      )
    });
  });
});

Enzyme.configure({ adapter: new Adapter() });
