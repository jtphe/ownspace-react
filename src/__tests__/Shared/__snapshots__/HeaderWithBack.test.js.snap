// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Header snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#003466",
      "flexDirection": "row",
      "paddingBottom": 30,
      "paddingTop": 40,
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      style={
        Object {
          "marginLeft": 30,
          "marginTop": 8,
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 38,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </RNGestureHandlerButton>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginLeft": 118,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
  </View>
</View>
`;
